src/braket/experimental/autoqasm/doc/decorators.md

```diff
@@ -0,0 +1,123 @@
+```
+@aq.subroutine
+def bell(q0: int, q1: int) -> None:
+    h(q0)
+    cnot(q0, q1)
+
+    
+@aq.main(num_qubits=4)
+def two_bell() -> None:
+    bell(0, 1)
+    bell(2, 3)
+
+two_bell_program = two_bell()
+```
+
+Let's take a look at the serialized output from `two_bell_program.to_ir()`, which shows that the modularity of the subroutine is preserved.
+
+```
+OPENQASM 3.0;
+def bell(int[32] q0, int[32] q1) {
+    h __qubits__[q0];
+    cnot __qubits__[q0], __qubits__[q1];
+}
+qubit[4] __qubits__;
+bell(0, 1);
+bell(2, 3);
+```
+
+## `@aq.gate`
+
+Represents a gate definition.
+
+Gate definitions define higher-level gates with support gates, and are often used to decompose a gate into the native gates of a device.
+
+The body of a gate definition can only contain gates. Qubits used in the body of a gate definition must be passed as input arguments, with the type hint `aq.Qubit`. Like subroutines, a gate must be called by a main quantum program to have an effect.
+
+```
+@aq.gate
+def ch(q0: aq.Qubit, q1: aq.Qubit):
+    """Define a controlled-Hadamard gate."""
+    ry(q1, -math.pi / 4)
+    cz(q0, q1)
+    ry(q1, math.pi / 4)
+    
+@aq.main(num_qubits=2)
+def main():
+    h(0)
+    ch(0, 1)
+    
+main_program = main()
+```
+
+
+## `@aq.gate_calibration`
+
+This decorator allows you to register a calibration for a gate. A gate calibration is a device-specific, low-level, pulse implementation for a logical gate operation.
+
+At the pulse level, qubits are no longer interchangable. Each one has unique properties. Thus, a gate calibration is usually defined for a concrete set of qubits and parameters, but you can use input arguments to your function as well.
+
+The body of a gate calibration must only contain pulse operations. This decorator requires one input arguments to specify the `Gate` that the calibration will be registered to. Concrete values for the qubits and parameters are supplied as keyword arguments to the decorator.
+The union of the arguments of the decorator and the decorated function must match the arguments of the gate to be implemented.
+
+For example, the gate `rx` takes two arguments, target and angle. Each arguments must be either set in the decorator or declared as an input parameter to the decorated function. To add the gate calibration to your program, use the `with_calibrations` method of the main program.
+
+```
+# This calibration only applies to physical qubit zero, so we
+# mark that in the decorator call
+@aq.gate_calibration(implements=rx, target="$0")
+def cal_1(angle: float):
+    # The calibration is applicable for any rotation angle,
+    # so we accept it as an input argument
+    pulse.barrier("$0")
+    pulse.shift_frequency(q0_rf_frame, -321047.14178613486)
+    pulse.play(q0_rf_frame, waveform(angle))
+    pulse.shift_frequency(q0_rf_frame, 321047.14178613486)
+    pulse.barrier("$0")
+    
+@aq.main
+def my_program():
+    rx(0, 0.123)
+    measure(0)
+```
```

Comment:
Could you also include the example of calling `with_calibrations()` to attach the calibration?